Added kafka consumer method

franciszekjob opened the pull request.

Commit:
added history_writer with kafka consumer

Commit:
added function description

Commit:
deleted value serializer in history_writer

Commit:
added api

Commit:
added api

amasend commented on api/app.py:
```diff
@@ -0,0 +1,33 @@
+
+
+# user creates a bot
+@app.route("/<apiKey>/bots", methods=["POST"])
```
`api key` should not be send via path but rather inside request header

amasend commented on api/app.py:
```diff
@@ -0,0 +1,33 @@
+
+
+# user gets his trade history
+@app.route("/<apiKey>/trade-history", methods=["GET"])
```
same as above

amasend commented on history_writer/history_writer.py:
```diff
@@ -0,0 +1,51 @@
+        enable_auto_commit=True)
+
+    for kafka_data in consumer:
+        data = kafka_data
```
Why we need another assignment here?

franciszekjob replied:
Fixed now, didn't push changes to the repo

amasend reviewed (changes requested):
PR has conflicts, please resolve them.
I see that API is mocked for now. We can accept this but in future this needs to have implementation.

Commit:
changed data to value kafka_data

Commit:
changed "/" for "_" in current data provider, in crypto symbol

Before: BTC/USDT
Now: BTC_USDT

Commit:
Update provider_manager.py

Commit:
added datatabase manager and test

franciszekjob committed:
Update app.py

franciszekjob committed:
Update app.py

Commit:
added add_exchange and curren_crypto_data

User now is able to add exchange and get current cryptocurrency data

Commit:
Merge branch 'downloader' of https://github.com/Fiiranek/crypto-auto-trader-platform into downloader

Commit:
added trade-history insight

Commit:
Merge remote-tracking branch 'upstream/dev' into downloader

Commit:
added test, fixed datat provider, created user auth method

Added API tests, user authentication method. fixed current data provider methods

amasend commented on api/database/database_manager.py:
```diff
@@ -0,0 +1,113 @@
+        symbol: str - symbol of cryptocurrency, e.g. BTC/USDT
+    """
+    user_data = authenticate_user(cursor, username, password)
+    if user_data != False:
```
Why we have such weird comparison here?

```suggestion
    if not user_data:
```

amasend commented on api/database/database_manager.py:
```diff
@@ -0,0 +1,113 @@
+            connection.commit()
+            return "Exchange created"
+        except:
+            return "Error"
```
It is always better to raise some exception when there is an error then returning a string.

amasend commented on api/database/database_manager.py:
```diff
@@ -0,0 +1,113 @@
+        returning user;""", (username, password, api_key))
+        connection.commit()
+        return "User created"
+    except (Exception, psycopg2.DatabaseError) as error:
```
Exception should be returned.

amasend commented on api/database/database_manager.py:
```diff
@@ -0,0 +1,113 @@
+            return user_data
+        else:
+            return "Username or password is incorrect"
+    except:
```
Same as above.

amasend commented on api/database/database_manager.py:
```diff
@@ -0,0 +1,113 @@
+        exchange.secret = secret
+        if exchange.has['fetchMyTrades']:
+            return exchange.fetch_my_trades(symbol=symbol)
+        return "Cannot get trades"
```
Raise an exception.

amasend commented on api/database/database_manager.py:
```diff
@@ -0,0 +1,113 @@
+            return exchange.fetch_my_trades(symbol=symbol)
+        return "Cannot get trades"
+    else:
+        return "Username or password is incorrect"
```
Same a above.

amasend reviewed (changes requested):
Read about exception handling in Python.
It is always better to create own exceptions (meaningfull one) and implement a leayer that handles them eg in API to return appropriate errors to user.

franciszekjob committed:
Update api/database/database_manager.py

Co-authored-by: Amadeusz Masny <amcia94@gmail.com>

Commit:
added custom exceptions

Commit:
resolved conflict

amasend reviewed (changes requested):
Please resolve conflicts and I will merge that PR.

franciszekjob committed:
Merge branch 'dev' into downloader
